src/app/models/treeoperations.cpp

```diff
@@ -157,3 +157,8 @@ void TreeOperations::notifyDbWasUnloaded(int dbIndex)
 {
     emit closeDbKeys(m_connection, dbIndex);
 }
+
```

Comment:
You shouldn't redirect this operation to ValueEditor, just remove key in TreeOperations model and send notification about removed key to ValueEditor::ViewModel.

Example here: https://github.com/uglide/RedisDesktopManager/blob/0.8.0/src/modules/connections-tree/items/databaseitem.cpp#L93-L103